src/EnergyPlus/SetPointManager.cc

```diff
@@ -5764,26 +6108,7 @@ void DefineOutsideAirSetPointManager::calculate(EnergyPlusData &state)
         SetTempAtOutHigh = this->OutHighSetPt1;
     }
 
-    this->SetPt = CalcSetPoint(OutLowTemp, OutHighTemp, state.dataEnvrn->OutDryBulbTemp, SetTempAtOutLow, SetTempAtOutHigh);
-}
-
-Real64 DefineOutsideAirSetPointManager::CalcSetPoint(
-    Real64 OutLowTemp, Real64 OutHighTemp, Real64 OutDryBulbTemp, Real64 SetTempAtOutLow, Real64 SetTempAtOutHigh)
-{
-    Real64 SetPt;
-    if (OutLowTemp < OutHighTemp) { // && SetTempAtOutLow > SetTempAtOutHigh
-        if (OutDryBulbTemp <= OutLowTemp) {
-            SetPt = SetTempAtOutLow;
-        } else if (OutDryBulbTemp >= OutHighTemp) {
-            SetPt = SetTempAtOutHigh;
-        } else {
-            SetPt = SetTempAtOutLow - ((OutDryBulbTemp - OutLowTemp) / (OutHighTemp - OutLowTemp)) * (SetTempAtOutLow - SetTempAtOutHigh);
-        }
-
-    } else {
-        SetPt = 0.5 * (SetTempAtOutLow + SetTempAtOutHigh);
-    }
-    return SetPt;
```

Comment:
since the same linear interpolation calculation is used in the new objects. We took this out of this function and made it independent, which prevented us from having duplicated codes.